src/gui/guiFormSpecMenu.cpp

```diff
@@ -1220,12 +1322,34 @@ void GUIFormSpecMenu::parseLabel(parserData* data, const std::string &element)
 			// than multiply by 0.4, to get exact results
 			// in the integer cases: 0.4 is not exactly
 			// representable in binary floating point.
-			s32 posy = pos.Y + ((float)i) * spacing.Y * 2.0 / 5.0;
 			std::wstring wlabel = utf8_to_wide(unescape_string(lines[i]));
-			core::rect<s32> rect = core::rect<s32>(
-				pos.X, posy - m_btn_height,
-				pos.X + m_font->getDimension(wlabel.c_str()).Width,
-				posy + m_btn_height);
+
+			core::rect<s32> rect;
+
+			if (data->real_coordinates) {
+				v2s32 pos = getRealCoordinateBasePos(false, &v_pos);
+
+				// Add offset for smaller rect.
+				pos.Y += (imgsize.Y / 4) + ( ((float)imgsize.Y) * i / 2);
+
+				rect = core::rect<s32>(
+					pos.X, pos.Y,
+					pos.X + m_font->getDimension(wlabel.c_str()).Width,
+					pos.Y + imgsize.Y / 2);
+
+			} else {
+				v2s32 pos = getElementBasePos(false, nullptr);
+				pos.X += stof(v_pos[0]) * spacing.X;
+				pos.Y += (stof(v_pos[1]) + 7.0f / 30.0f) * spacing.Y;
+
+				s32 posy = pos.Y + ((float)i) * spacing.Y * 2.0 / 5.0;
```

Comment:
There's no reason for an extra variable anymore.
```suggestion
				pos.Y += ((float) i) * spacing.Y * 2.0 / 5.0;
```
(Also change `posy` to `pos.Y` below.)